timeflux_dsp/nodes/spectral.py

```diff
@@ -63,36 +64,50 @@ def __init__(self, fs=1.0, nfft=None, return_onesided=True):
                                               If `False` return a two-sided spectrum.
                                               (Note that for complex data, a two-sided spectrum is always returned.)
                                               Default: `True`.
+                interpolation (bool|dict): How to interpolate NaN values in the input using Pandas interpolate.
         """
 
-        self._fs = fs
-        self._nfft = nfft
+        self._fixed_rate = rate
+        self._rate = rate
+        self._nfft = None if nfft is None else int(nfft)
+        self._fixed_nfft = self._nfft
+
+        self._interpolation = {
+            'method': 'time',
+            'limit': 10,
+        }
+        if type(interpolation) is dict:
+            self._interpolate = True
+            self._interpolation = {**self._interpolation, **interpolation}
+        else:
+            self._interpolate = interpolation
+
         if return_onesided:
             self._sides = 'onesided'
         else:
             self._sides = 'twosided'
-        if self._nfft is not None:
+        if self._nfft is not None and self._rate is not None:
             self._set_freqs()
 
     def _check_nfft(self):
 
         # Check validity of nfft at first chunk
-        if self._nfft is None:
+        if self._fixed_nfft is None:
             self.logger.debug("nfft := length of the chunk ")
             self._nfft = self.i.data.shape[0]
             self._set_freqs()
-        elif self._nfft < self.i.data.shape[0]:
-            raise ValueError('nfft must be greater than or equal to length of chunk.')
+        elif self._fixed_nfft > self.i.data.shape[0]:
+            raise ValueError('nfft must be less than or equal to length of chunk.')
         else:
-            self._nfft = int(self._nfft)
+            self.o.data = self.o.data.tail(self._fixed_nfft)
```

Comment:
I don't like tailing the data. If such a thing happen, the user should be at least warned.

Reply:
The performance with FFT is better if you keep to powers of 2 like 64, 128, 256.  https://stackoverflow.com/questions/46357589/numpy-fft-is-fast-for-lengths-that-are-products-of-small-primes-but-how-small But it's not simple to provide a window that delivers exactly 256 samples

Reply:
It seems easier and safer to get the window to give a sensible round number of seconds like 1.5 or 2, then getting the FFT to truncate to 256 samples, rather than calculating that your window needs to be 1.234678 seconds long and trusting the Window node to give you the result you expect.

Reply:
We could of course add a max_samples argument to the Window node and give it responsibiity for this.

Reply:
I understand. I have a node for that purpose, `FixedWindow`, where you specify length with samples instead of seconds. I'll push it tomorrow in module `timeflux.nodes.window`. 
Here, tailing will make the data discontinuous. Imagine on a sinus signal!

Reply:
OK, removed